### PR TITLE
DNM: Wip bs batch io completion

### PR DESCRIPTION
**Pull request by aclamk**
Highly experimental work, it depends on rzarzynski batching work.
It is provided to make performance tests of bluestore.

### CODE REVIEW

File: src/os/bluestore/BlueStore.h [hunk 1]
```diff
@@ -1673,6 +1745,26 @@ class BlueStore : public ObjectStore,
     }
   };
 
+  struct NagleBatcherThread : public Thread {
+    BlueStore* const store;
```
Comment (amitkumar50) [inline src/os/bluestore/BlueStore.h]: @aclamk 
https://google.github.io/styleguide/cppguide.html#Constant_Names

File: src/os/bluestore/BlueStore.h [hunk 2]
```diff
@@ -1517,6 +1519,76 @@ class BlueStore : public ObjectStore,
...
+      for (auto txc : *this) {
+        txc->state = new_state;
+      }
+      state = new_state;
+    }
+
+    TransBatch dissect(std::function<bool(TransContext*)> pred) {
+      TransBatch dissected(get_state());
+
+      for (auto it = this->begin(); it != this->end(); /* see body */) {
```
Comment (amitkumar50) [inline src/os/bluestore/BlueStore.h]: @aclamk How about range-based for loop c++11 here.
Your thoughts?

File: src/os/bluestore/KernelDevice.cc [hunk 3]
```diff
@@ -275,10 +276,13 @@ void KernelDevice::_aio_thread()
 	  void *priv = ioc->priv;
 	  ioc->aio_wake();
 	  if (priv) {
-	    aio_callback(aio_callback_priv, priv);
+            batch.push_back(priv);
```
Comment (amitkumar50) [inline src/os/bluestore/KernelDevice.cc]: @aclamk Can't we remove this tab space if it was present in older commits also?

File: src/os/bluestore/KernelDevice.cc [hunk 4]
```diff
@@ -392,39 +396,27 @@ void KernelDevice::aio_submit(IOContext *ioc)
   ioc->num_pending -= pending;
   assert(ioc->num_pending.load() == 0);  // we should be only thread doing this
 
-  bool done = false;
-  while (!done) {
-    FS::aio_t& aio = *p;
+  for (auto it = p; it != e; it++) {
```
Comment (amitkumar50) [inline src/os/bluestore/KernelDevice.cc]: @aclamk How about using range-based for loop here, that's my opinion..
Your thoughts?

File: src/os/bluestore/BlueStore.h (hunk 2 above)
Comment (aclamk) [inline src/os/bluestore/BlueStore.h]: @amitkumar50 Any of those cosmetic changes can be made. However, this PR was made half a year ago. I am sure that there are more several problems with current master.

**Pull request closed without merging**